independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java

```diff
@@ -197,18 +197,47 @@ public static DefaultRepositorySystemSession newSession(RepositorySystem system,
 
     public static List<RemoteRepository> getRemoteRepos(Settings settings) throws AppModelResolverException {
         final List<RemoteRepository> remotes = new ArrayList<>();
+        // first it's the active by default ones
         for (Profile profile : settings.getProfiles()) {
             if (profile.getActivation() != null && profile.getActivation().isActiveByDefault()) {
                 addProfileRepos(profile, remotes);
             }
         }
+        // then it's  the ones activated with -P
+        if(mvnCmd != null) {
```

Comment:
Hmmm. That seems very fragile. There's no other way to do that?

BTW, Maven supports multiple `-P` options.

Reply:
> Hmmm. That seems very fragile. There's no other way to do that?

I couldn't find any alternative so far.
Imagine you are writing a JUnit test and in the test you need to figure out the Maven config that launched that test (also it's possible that the test could have been launched not from Maven).
That essentially means that all that's available for us to discover is system props and env. There is also classpath and filesystem though but not sure that could help much?

> BTW, Maven supports multiple `-P` options.

Thanks! Added support for multiple `-P`'s,